internal/kafka/internal/config/dynamic_scaling_config_test.go

```diff
@@ -0,0 +1,326 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/onsi/gomega"
+)
+
+func TestDynamicScalingComputeNodesConfig_validate(t *testing.T) {
+	t.Parallel()
+	type fields struct {
+		MaxComputeNodes int
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		wantErr bool
+	}{
+		{
+			name: "should return an error when MaxComputesNodes is less than 1",
+			fields: fields{
+				MaxComputeNodes: 0,
+			},
+			wantErr: true,
+		},
+		{
+			name: "shouldn't return an error when MaxComputesNodes is greater than 0",
+			fields: fields{
+				MaxComputeNodes: 3,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		testcase := tt
+		t.Run(testcase.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+			t.Parallel()
+			c := &DynamicScalingComputeNodesConfig{
+				MaxComputeNodes: testcase.fields.MaxComputeNodes,
+			}
+
+			err := c.validate()
+			g.Expect(err != nil).To(gomega.Equal(testcase.wantErr))
```

Comment:
```suggestion
			g.Expect(c.validate() != nil).To(gomega.Equal(testcase.wantErr))
```

This pattern could possibly be applied across the whole test file, where single value is returned by a tested function

Reply:
Thanks, I prefer to the act-and-assert test pattern which separate the `ACT` where the output of the action is grabbed, and it is separated from the `ASSERT` part of the test where one or several assertions are performed.

Reply:
Thanks.

Can you give it a lgtm and we can get this in?